packages/twenty-server/src/database/commands/0-22-add-new-address-field-to-views-with-deprecated-address.command.ts

```diff
@@ -0,0 +1,225 @@
+import {
+  FeatureFlagEntity,
+  FeatureFlagKeys,
+} from 'src/engine/core-modules/feature-flag/feature-flag.entity';
+import { Workspace } from 'src/engine/core-modules/workspace/workspace.entity';
+import { DataSourceService } from 'src/engine/metadata-modules/data-source/data-source.service';
+import { FieldMetadataEntity } from 'src/engine/metadata-modules/field-metadata/field-metadata.entity';
+import { WorkspaceCacheVersionService } from 'src/engine/metadata-modules/workspace-cache-version/workspace-cache-version.service';
+import { TwentyORMManager } from 'src/engine/twenty-orm/twenty-orm.manager';
+import { COMPANY_STANDARD_FIELD_IDS } from 'src/engine/workspace-manager/workspace-sync-metadata/constants/standard-field-ids';
+import { ViewFieldWorkspaceEntity } from 'src/modules/view/standard-objects/view-field.workspace-entity';
+
+interface AddNewAddressFieldToViewsWithDeprecatedAddressFieldCommandOptions {
+  workspaceId?: string;
+}
+
+@Command({
+  name: 'migrate-0.22:add-new-address-field-to-views-with-deprecated-address-field',
+  description: 'Adding new field Address to views containing old address field',
+})
+export class AddNewAddressFieldToViewsWithDeprecatedAddressFieldCommand extends CommandRunner {
+  private readonly logger = new Logger(
+    AddNewAddressFieldToViewsWithDeprecatedAddressFieldCommand.name,
+  );
+  constructor(
+    @InjectRepository(Workspace, 'core')
+    private readonly workspaceRepository: Repository<Workspace>,
+    @InjectRepository(BillingSubscription, 'core')
+    private readonly billingSubscriptionRepository: Repository<BillingSubscription>,
+    @InjectRepository(FeatureFlagEntity, 'core')
+    private readonly featureFlagRepository: Repository<FeatureFlagEntity>,
+    @InjectRepository(FieldMetadataEntity, 'metadata')
+    private readonly fieldMetadataRepository: Repository<FieldMetadataEntity>,
+    private readonly typeORMService: TypeORMService,
+    private readonly dataSourceService: DataSourceService,
+    private readonly workspaceCacheVersionService: WorkspaceCacheVersionService,
+    private readonly twentyORMManager: TwentyORMManager,
+  ) {
+    super();
+  }
+
+  @Option({
+    flags: '-w, --workspace-id [workspace_id]',
+    description: 'workspace id. Command runs on all workspaces if not provided',
+    required: false,
+  })
+  parseWorkspaceId(value: string): string {
+    return value;
+  }
+
+  async run(
+    _passedParam: string[],
+    options: AddNewAddressFieldToViewsWithDeprecatedAddressFieldCommandOptions,
+  ): Promise<void> {
+    this.logger.log('running');
+    let workspaceIds: string[] = [];
+
+    if (options.workspaceId) {
+      workspaceIds = [options.workspaceId];
+    } else {
+      const workspaces = await this.workspaceRepository.find();
+
+      const activeWorkspaceIds = (
+        await Promise.all(
+          workspaces.map(async (workspace) => {
+            const isActive = await this.workspaceIsActive(workspace);
```

Comment:
What is the downside of performing the change on inactive workspaces besides the time to run this query? I'd say that as long as they have an existing workspace schema we should apply the migration. As it might be re-enabled?

Reply:
The downside is that inactive workspaces are not being synchronized with sync-metadata anymore, and may have subsequent issues making the command hard to apply or not applicable on them. We had the issues yesterday when running the command for the boolean fields on all the workspaces (= +8k while we have 375 active ones), at first I started fixing them but it was too time consuming and mostly time wasted on unused workspaces, I ended up only running the command for the active workspaces which was a bit painful too (I needed to run individual commands for each workspace id then to to execute lines 40 by 40)